src/controllers/certificates/certs.go

```diff
@@ -101,23 +103,11 @@ func (cs *Certs) validateServerCerts(now time.Time) bool {
 		return true
 	}
 
-	block, _ := pem.Decode(cs.Data[ServerCert])
-	if block == nil {
-		log.Info("failed to parse server certificates, renewing", "error", "can't decode PEM file")
-		return true
-	}
-
-	cert, err := x509.ParseCertificate(block.Bytes)
-	if err != nil {
-		log.Info("failed to parse server certificates, renewing", "error", err)
+	isValid, err := kubeobjects.ValidateCertificateExpiration(cs.Data[ServerCert], renewalThreshold, now, log)
```

Comment:
You could use this validation function in `validateRootCert` too

Reply:
Sadly not, as `validateRootCert` sets also the parsed certificate to `cs.rootPublicCert`, which would not happen if I only call my function. Generating a new serverCert would fail if a valid root certificate is there, but cs.rootPublicCert is not set.

Reply:
can we change the logic on either side so we reuse as much functionality as possible?

Reply:
I guess i could refactor more here, wanted to avoid that in order to fix the bug. But could make a function which returns the certificate and a seperated one, which validates the cert validity